source/shared/cpp/ObjectModel/ParseUtil.cpp

```diff
@@ -54,6 +54,20 @@ namespace AdaptiveSharedNamespace
         }
     }
 
+    std::string ParseUtil::TryGetString(const Json::Value& json, AdaptiveCardSchemaKey key)
+    {
+        std::string propertyName = AdaptiveCardSchemaKeyToString(key);
+        auto propertyValue = json.get(propertyName, Json::Value());
+        if (propertyValue.empty() || !propertyValue.isString())
+        {
+            return "";
+        }
+        else
+        {
+            return propertyValue.asString();
+        }
```

Comment:
isn't this just `ParseUtil::GetString(json, key, "", false);`?

Reply:
ParseUtil::GetString(json, key, "", false); will throw if the value at key exists but is not a string. In the case this is handling, the value at key may very well be an object.

---
In reply to: [271454369](https://github.com/Microsoft/AdaptiveCards/pull/2609#discussion_r271454369) [](ancestors = 271454369)

Reply:
(the shared model changes here are at least mostly from my highlight change that just went in, and won't show up once this is updated to the latest master)

---
In reply to: [271827206](https://github.com/Microsoft/AdaptiveCards/pull/2609#discussion_r271827206) [](ancestors = 271827206,271454369)